File: src/main/java/com/DavidM1A2/AfraidOfTheDark/common/constants/ModBoltEntries.java
package com.DavidM1A2.afraidofthedark.common.constants;

import com.DavidM1A2.afraidofthedark.common.entity.bolt.EntityIronBolt;
import com.DavidM1A2.afraidofthedark.common.entity.bolt.EntityWoodenBolt;
import com.DavidM1A2.afraidofthedark.common.registry.bolt.AOTDBoltEntry;
import com.DavidM1A2.afraidofthedark.common.registry.bolt.BoltEntry;

/**
 * A static class containing all of our bolt entry references for us
 */
public class ModBoltEntries
{
	public static final BoltEntry WOODEN = new AOTDBoltEntry("wooden", ModItems.WOODEN_BOLT, EntityWoodenBolt::new, null);
	public static final BoltEntry IRON = new AOTDBoltEntry("iron", ModItems.IRON_BOLT, EntityIronBolt::new, null);

	public static final BoltEntry[] BOLT_ENTRY_LIST = new BoltEntry[]
	{
		WOODEN,
		IRON
	};
}

File: src/main/java/com/davidm1a2/afraidofthedark/common/constants/ModBoltEntries.kt
package com.davidm1a2.afraidofthedark.common.constants

import com.davidm1a2.afraidofthedark.common.entity.bolt.*
import com.davidm1a2.afraidofthedark.common.registry.bolt.AOTDBoltEntry

/**
 * A static class containing all of our bolt entry references for us
 */
object ModBoltEntries {
    val WOODEN = AOTDBoltEntry(
        "wooden",
        ModItems.WOODEN_BOLT,
        { world, thrower -> EntityWoodenBolt(thrower, world) },
        null
    )
    val IRON = AOTDBoltEntry(
        "iron",
        ModItems.IRON_BOLT,
        { world, thrower -> EntityIronBolt(thrower, world) },
        null
    )
    val ASTRAL_SILVER = AOTDBoltEntry(
        "astral_silver",
        ModItems.SILVER_BOLT,
        { world, thrower -> EntitySilverBolt(thrower, world) },
        ModResearches.ASTRAL_SILVER
    )
    val IGNEOUS = AOTDBoltEntry(
        "igneous",
        ModItems.IGNEOUS_BOLT,
        { world, thrower -> EntityIgneousBolt(thrower, world) },
        ModResearches.IGNEOUS
    )
    val STAR_METAL = AOTDBoltEntry(
        "star_metal",
        ModItems.STAR_METAL_BOLT,
        { world, thrower -> EntityStarMetalBolt(thrower, world) },
        ModResearches.STAR_METAL
    )

    val BOLT_ENTRY_LIST = arrayOf(
        WOODEN,
        IRON,
        ASTRAL_SILVER,
        IGNEOUS,
        STAR_METAL
    )
}